spec/Failsafe.spec.ts

```diff
@@ -171,6 +171,126 @@ describe(`Failsafe`, function() {
             expect(logger.stdout()).to.include(`[check-ansi-support] Colors not supported`);
         });
     });
+
+    describe('Pass arguments', () => {
+
+        it(`fails on unknown arguments`, async () => {
+            const { run, logger } = failsafe();
+
+            const exitCode = await run([`print-args`, '--foo=bar', '--bar=foo']);
+            expect(exitCode).to.equal(General_Failure, `Expected exit code of ${General_Failure}${ format(logger) }`);
+
+            expect(logger.stderr()).to.include(`Unknown argument '--foo=bar'`);
+        });
+
+        it(`passes specific arguments`, async () => {
+            const { run, logger } = failsafe();
+
+            const exitCode = await run([`print-args`, '[--foo]', '--foo=bar']);
+            expect(exitCode).to.equal(Success, `Expected exit code of ${Success}${ format(logger) }`);
+
+            expect(logger.stdout()).to.include([
+                `[print-args] Listing 1 arguments`,
+                `[print-args] --foo=bar`,
+                `[failsafe] Script 'print-args' exited with code 0`,
+            ].join('\n'));
+        });
+
+        it(`passes specific arguments interleaved`, async () => {
+            const { run, logger } = failsafe();
+
+            const exitCode = await run(['print-args[--foo][--bar]', '--foo', '--bar']);
+            expect(exitCode).to.equal(Success, `Expected exit code of ${Success}${ format(logger) }`);
+
+            expect(logger.stdout()).to.include([
+                `[print-args] Listing 2 arguments`,
+                `[print-args] --foo`,
+                `[print-args] --bar`,
+                `[failsafe] Script 'print-args' exited with code 0`,
+            ].join('\n'));
+        });
+
+        it(`passes all arguments`, async () => {
+            const { run, logger } = failsafe();
+
+            const exitCode = await run([`print-args`, '[...]', '--foo=bar', '--bar=foo']);
```

Comment:
I like the idea of the wildcard argument; I wonder though how exactly it should behave?
It _feels_ like `...` should mean "any argument, not explicitly named", so a "rest" operator of sorts. 

For example, given scripts are defined as follows:
```
test: "failsafe clean[--force] test[--name,--spec] report"
```
I'd expect that the below invocation:
```
npm test -- --force --name="my test"
```
should result in:
- `--force` flag passed to the `clean` script
- `--name` parameter passed to `test` script

However, if I invoked the script with an unexpected parameter, like `--suite`
```
npm test -- --force --suite="my suite"
```
I'd expect `failsafe` to fail complaining that `--suite` is not an expected parameter,
it should tell the developer what the expected parameters are, and offer them to use a wildcard if needed

With a wildcard defined say like this:
```
test: "failsafe clean[--force] test[--name,--spec,...] report"
```
I'd expect failsafe to pass `--force` to `clean`, `--name` and `--spec` to `test`, and not complain on unexpected, but instead pass those other parameters to `test`

Is that what you had in mind?

Reply:
Of course, there's also the question of what happens when there are multiple wildcards, but I think that's fine and we'd need to pass the "unexpected" parameters to any script that accepts the wildcard
```
test: "failsafe clean[--force] test[--name,--spec,...] log-parameters[...] report"
```

Reply:
I agree with all you’ve written. And yes we should support passing the same argument to multiple scripts. And that also includes the wildcard arg “…”.

we should also allow varying whitespace in these definitions… so the parsing will be a bit more complex. But I think it will be worth it. :-)